src/Controls/src/Core/BindableObject.cs

```diff
@@ -253,8 +253,19 @@ public void RemoveBinding(BindableProperty property)
 		{
 			BindablePropertyContext context = GetContext(property ?? throw new ArgumentNullException(nameof(property)));
 
-			if (context?.Binding != null)
-				RemoveBinding(property, context);
+			var specificity = SetterSpecificity.FromBinding;
+			if (context != null && context.Bindings.Count > 0)
+				specificity = context.Bindings.Last().Key;
+
+			RemoveBinding(property, specificity);
+		}
+
+		internal void RemoveBinding(BindableProperty property, SetterSpecificity specificity)
```

Comment:
Should this public for Library authors to use ?

Reply:
no, specificity isn't public